persistence/src/main/java/org/mskcc/smile/persistence/neo4j/SmileSampleRepository.java

```diff
@@ -132,6 +132,12 @@ SmileSample updateRevisableBySampleId(@Param("smileSampleId") UUID smileSampleId
     List<SmileSample> findSamplesByCohortId(@Param("cohortId") String cohortId);
 
     @Query("MATCH (sm:SampleMetadata {primaryId: $primaryId})"
-            + "RETURN sm.cmoSampleName ORDER BY sm.importDate DESC LIMIT 1")
-    String findCmoSampleNameByPrimaryId(@Param("primaryId") String primaryId);
+            + "RETURN sm ORDER BY sm.importDate DESC LIMIT 1")
+    SampleMetadata findLatestSampleMetadataByPrimaryId(@Param("primaryId") String primaryId);
+
+    @Query("MATCH (pa:PatientAlias {namespace: $patientAliasType})-[:IS_ALIAS]->(p:Patient)-[:HAS_SAMPLE]->"
```

Comment:
patient repository might already have this method and should be accessible to the smile sample service